scripts/synthetix/facilitate_trade.js

```diff
@@ -121,33 +117,47 @@ module.exports = async (callback) => {
     const ourBuyPrice = formatedRate * (1 - sUSDTosETHFee)
     if (ourBuyPrice > theirSellPrice) {
       // We are willing to pay more than the exchange is selling for.
-      console.log(`Placing an order to buy sETH at ${ourBuyPrice}`)
-      const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
-        1 / ourBuyPrice,
-        sETH.decimals,
-        sUSD.decimals
-      )
-      orders.push({
-        buyToken: sETH.exchangeId,
-        sellToken: sUSD.exchangeId,
-        buyAmount: buyETHAmount,
-        sellAmount: sellSUSDAmount,
-      })
+      console.log(`Placing an order to buy sETH at ${ourBuyPrice}, but verifying sUSD balance first`)
+      const sUSDBalance = await exchange.getBalance(account, sUSD.address)
+      if (sUSDBalance >= minSellsUSD) {
+        const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
+          1 / ourBuyPrice,
+          sETH.decimals,
+          sUSD.decimals
+        )
+        orders.push({
+          buyToken: sETH.exchangeId,
+          sellToken: sUSD.exchangeId,
+          buyAmount: buyETHAmount,
+          sellAmount: sellSUSDAmount,
+        })
+      } else {
+        console.log(`Warning: Insufficient sUSD (${sUSDBalance} < ${minSellsUSD}) for order placement.`)
+      }
```

Comment:
I'd consider moving line 120 inside the if and remove "but verifying sUSD balance first", unless you really want to stress that this check is being done in the log. Same change in the if below.

Reply:
I want the user to know that an preset was being attempted. But that the balance failed. If the first log is moved inside of the balance verification the user might not know what happened.